### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by sagitta42**
In this PR: documentation including manuals and tutorials on field / ideal pulse / realistic waveform simulation.
Tested locally on MPIK with html preview.

Small fixes:
- added wrapper to launch `simulate_fields` providing detector metadata `PropDict` directly as an option (and not only path to JSON). Feature needed by Simon Sailer for crystal cut simulations convenience
- fixed bug in vendor dead layer check (`PropDicts.MissingProperty` instead of `isnothing`) in SSD and siggen (Note: maybe this check should happen beforehand once and not separately twice in SSD and siggen geometry...)

### CODE REVIEW

File: src/detector.jl [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
+function simulate_fields(detector_metadata::Union{AbstractString, PropDict}, all_settings::Union{Dict,PropDict}; overwrite::Bool = false)
+    simulate_fields(detector_metadata, PropDict(all_settings).environment, PropDict(all_settings).simulation; overwrite)
 end
 
 # user launches with all settings in json
-function simulate_fields(detector_metadata_path::AbstractString, all_settings::AbstractString; overwrite::Bool = false)
-    simulate_fields(detector_metadata_path, propdict(all_settings); overwrite)
+function simulate_fields(detector_metadata::Union{AbstractString, PropDict}, all_settings::AbstractString; overwrite::Bool = false)
+    simulate_fields(detector_metadata, propdict(all_settings); overwrite)
 end
```
Comment (fhagemann) [inline src/detector.jl:27]: I am not too familiar with `PropDicts.jl`, is it correct that it is `PropDict(all_settings)` (uppercase)  in the upper one but `propdict(all_settings)` (lowercase) in the lower one?

Comment (sagitta42) [reply src/detector.jl:27]: So `PropDict(all_settings)` is for `all_settings::Dict` (if it's a `PropDict` already, nothing happens); and `LegendGeSim.propdict(all_settings)` is for `all_settings::AbstractString` (path to JSON). This is an obsolete function in `temp_utils.jl`, now it can be replaced with `readprops` from `PropDicts.jl` itself.

**Comment by fhagemann** [comment]
I didn't look through all steps of the documentation, but the build seems to pass so I guess this should be good to merge!

**Comment by oschulz** [comment]
Let's merge it then. :-)

**Comment by fhagemann** [comment]
There seems to be some issue when trying to deploy the docs. Can it be that the files created when building the documentation exceed the maximum file size on GitHub which prevents the deployment? @oschulz

**Comment by oschulz** [comment]
There are errors like

```dev/tutorials/cache/Public Inverted Coax_test_ssd_fields.h5f is 99.84 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB```

We must not include such files in the docs, not only are they beyond GitHub limits, they would also blow up the size of the repo immensely..

**Comment by fhagemann** [comment]
Then I guess we shouldn't cache any simulation results when building the documentation

**Comment by oschulz** [comment]
> Then I guess we shouldn't cache any simulation results when building the documentation

Or just delete them at the end in a code block tagged for docs-only.

**Comment by fhagemann** [comment]
I made one PR where nothing is cached (#55) and one PR where the cache is deleted before deploying the docs (#56).
@sagitta42 @oschulz Feel free to choose whichever you like better and close the other.

**Comment by sagitta42** [comment]
@fhagemann @oschulz Firstly, I noticed a bug. In the Field Simulation tutorial, the simulation with cached name "test" is done for SSD with crystal impurity input for the V99000 detector. Later in Ideal Pulse Simulation as well as Realistic Waveform Simulation tutorials, an SSD simulation with cached name "test" is done for the Public Inv Coax (because there is PET input for this one). The simulation would then read the cached simulation that is for a different detector entirely.

Seeing that there is no need to cache the Field Simulation for later, as it's not used in the other tutorials, definitely I would remove caching there.

As for the PSS and raw simulation, we might want to cache in the PSS tutorial, so that it can be re-used in the real wf tutorial. Although, it's not like they are built in order like that? In that case, we can just not cache anything, and the detector simulation for both tutorials can run separately. I guess build time is not really of an issue right?

So in conclusion I would then go for #55 not caching anything.
